### PR TITLE
Support gzip and bz2 compression

### PR DESCRIPTION
**Pull request by Limess**
This change adds the .bz2 or .gzip file extension to files in S3 and locally, and uses the GZIP or BZ2 directive in the COPY command.

Configurable by the 'compression' parameter.

See https://docs.aws.amazon.com/redshift/latest/dg/copy-parameters-file-compression.html#copy-bzip2 for reference.

Stitch ran a benchmark of the performance of different compression formats a few years ago: https://www.stitchdata.com/blog/redshift-database-benchmarks-copy-performance-with-compressed-files/. This suggested that gzip/bz2 was a net positive over a certain filesize, but both are worse than LZO for small-medium file sizes, and both worse than zero compression for small file sizes.

### CODE REVIEW

File: target_redshift/__init__.py [hunk 1]
```diff
@@ -340,15 +342,31 @@ def load_stream_batch(stream, records_to_load, row_count, db_sync, delete_rows=F
         row_count[stream] = 0
 
 
-def flush_records(stream, records_to_load, row_count, db_sync):
-    csv_fd, csv_file = mkstemp()
-    with open(csv_fd, 'w+b') as f:
+def flush_records(config, stream, records_to_load, row_count, db_sync):
```
Comment (koszti) [inline target_redshift/__init__.py]: list of arguments changed by adding `config`arg but only here and not at the places where it's been called.

**Comment by koszti** [comment]
@Limess thanks for this, this is really great. However I found something that's not allowing me to run it. I commented at the right place and the fix is looking straightforward.

I'm looking forward to merge it.

**Comment by Limess** [comment]
Ah sorry about that, I'd extracted the merged changes from a fork into two separate changes and forgot about updating the config handling. I'll fix that tomorrow on both PRs if I have time.

**Comment by koszti** [comment]
thanks you. Now it's working but has some conflicts after #33 got merged. Can you please help resolving them?

**Comment by Limess** [comment]
I've rebased this branch.

**Pull request merged**